pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -240,6 +243,37 @@ func getLoadbalancerByName(client *gophercloud.ServiceClient, name string) (*loa
 	return &loadbalancerList[0], nil
 }
 
+func getSecurityGroupName(clusterName string, service *api.Service) string {
+	return fmt.Sprintf("lb-sg-%s-%v", clusterName, service.Name)
+}
+
+func getSecurityGroupRules(client *gophercloud.ServiceClient, opts rules.ListOpts) ([]rules.SecGroupRule, error) {
+
+	pager := rules.List(client, opts)
+
+	var securityRules []rules.SecGroupRule
+
+	err := pager.EachPage(func(page pagination.Page) (bool, error) {
+		ruleList, err := rules.ExtractRules(page)
+		if err != nil {
+			return false, err
+		}
+
+		for _, rule := range ruleList {
```

Comment:
Extremely minor point, but you can append lists with:
    securityRules = append(securityRules, ruleList...)